nixos/modules/programs/espanso-capdacoverride/espanso-capdacoverride.nix

```diff
@@ -0,0 +1,60 @@
+{
+  autoPatchelfHook,
+  capDacOverrideWrapperDir,
+  espanso,
+  patchelfUnstable, # have to use patchelfUnstable to support --rename-dynamic-symbols
+  stdenv,
+}:
+let
+  inherit (espanso) version;
+  pname = "${espanso.pname}-capdacoverride";
+
+  wrapperLibName = "wrapper-lib.so";
+  wrapperLibSource = "wrapper-lib.c";
+
+  # On Wayland, Espanso requires the DAC_OVERRIDE capability. One can create a wrapper binary with this
+  # capability using the `config.security.wrappers.<name>` framework. However, this is not enough: the
+  # capability is required by a worker process of Espanso created by forking `/proc/self/exe`, which points
+  # to the executable **without** the DAC_OVERRIDE capability. Thus, we inject a wrapper library into Espanso
+  # that redirects requests to `/proc/self/exe` to the binary with the proper capabilities.
```

Comment:
We can also just patch the code to point to our correct path which IMO sounds a lot less troublesome than all of this.

Reply:
Indeed, that's an alternative I thought about in the beginning. My main motivation for going this route was that it seems lower maintenance wrt upstream changes. With the wrapper library, no matter which of the Espanso subcommands perform a fork (now or sometime in the future), it is always caught and redirected. Patching the code would require manual updates.

I should also mention that, ideally, such an automatic fork redirect or something similar would be handled internally by `security.wrappers`. The reason that's currently not working has to do with the way how Espanso yields the capabilities early and relies on file capabilities to regain them when required (see [this comment](https://github.com/NixOS/nixpkgs/pull/328890#issuecomment-2265028930)).